### PR TITLE
planner: use code-gen to generate CloneForPlanCache method for Insert/Update/Delete

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #54057

Problem Summary: planner: use code-gen to generate CloneForPlanCache method for Insert/Update/Delete

### What changed and how does it work?

planner: use code-gen to generate CloneForPlanCache method for Insert/Update/Delete

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/planner/core/common_plans.go [hunk 1]
```diff
@@ -429,16 +436,19 @@ type Update struct {
 
 	SelectPlan base.PhysicalPlan
 
-	TblColPosInfos TblColPosInfoSlice
+	// TblColPosInfos is for multi-table update statement.
```
Comment (time-and-fate) [inline pkg/planner/core/common_plans.go:439]: I think single-table update will also use this field.

**Review by time-and-fate — APPROVED** [review]
others lgtm

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**